frontends/web/src/components/confirm/Confirm.tsx

```diff
@@ -42,27 +52,24 @@ export default class Confirm extends Component {
         this.setState({
             active: false,
             message: '',
-        }, () => {
-            this.callback = null;
         });
     }
 
-    render({}, {
+    public render({}, {
         message,
-        callback,
         active,
     }) {
-        if (!active) return null;
-        return (
-            <div class={style.overlay}>
-                <div class={style.confirmWindow}>
-                    <p>{message}</p>
-                    <div class={style.buttons}>
-                        <Button secondary onClick={() => this.respond(false)}>Cancel</Button>
-                        <Button primary onClick={() => this.respond(true)}>Confirm</Button>
-                    </div>
+        return active ? (
+            <Dialog>
+                <p class="first">{message}</p>
+                <div class="buttons flex flex-row flex-between">
+                    <Button secondary onClick={() => this.respond(false)}>Cancel</Button>
```

Comment:
Ideally, we would avoid creating new functions on every render (but I know that we still do it all over the place).